llvm/include/llvm/IR/IRBuilder.h

```diff
@@ -229,8 +239,11 @@ class IRBuilderBase {
   /// not on \p Src will be dropped from MetadataToCopy.
   void CollectMetadataToCopy(Instruction *Src,
                              ArrayRef<unsigned> MetadataKinds) {
-    for (unsigned K : MetadataKinds)
+    for (unsigned K : MetadataKinds) {
       AddOrRemoveMetadataToCopy(K, Src->getMetadata(K));
+      if (K == LLVMContext::MD_dbg)
+        SetCurrentDebugLocation(Src->getDebugLoc());
+    }
```

Comment:
Why this?

Reply:
`AddOrRemoveMetadataToCopy` only takes the `MDNode*` as an argument, so we can't capture the metadata from the source DebugLoc. We still call `AddOrRemoveMetadataToCopy` because this whole bit in IRBuilder is a bit messy; we want `MD_dbg` to still appear in the `MetadataToCopy` map because otherwise we won't know that we need to copy the stored debugloc to any inserted instruction.